// nestjs-typeorm-custom-repository/dist/custom-repository.module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomRepositoryModule = void 0;
const typeorm_1 = require("@nestjs/typeorm");
const custom_repository_util_1 = require("./utils/custom-repository.util");
function getProviders(repositories, dataSource) {
    return repositories.map((repository) => {
        const entity = (0, custom_repository_util_1.getEntityByRepository)(repository);
        return {
            module: CustomRepositoryModule,
            provide: (0, custom_repository_util_1.getCustomRepositoryToken)(repository, dataSource),
            useFactory: (dataSource) => {
                return new repository(entity, dataSource.manager);
            },
            inject: [(0, typeorm_1.getDataSourceToken)(dataSource)],
        };
    });
}
class CustomRepositoryModule {
    static forFeature(repositories, dataSource) {
        return {
            module: CustomRepositoryModule,
            providers: getProviders(repositories, dataSource),
            exports: getProviders(repositories, dataSource),
        };
    }
}
exports.CustomRepositoryModule = CustomRepositoryModule;
